nodejs/awsx/cloudwatch/metric.ts

```diff
@@ -188,29 +205,29 @@ export class Metric {
      */
     public withDimensions(dimensions: pulumi.Input<Record<string, any>> | undefined) {
         return new Metric({
-            ...this,
+            ...this.spread(),
```

Comment:
ts doesn't like spreading classes.  Fixes:  error TS2698: Spread types may only be created from object types.